app/plugins/benchmark_mode/benchmark_mode.cpp

```diff
@@ -0,0 +1,59 @@
+/* Copyright (c) 2020-2021, Arm Limited and Contributors
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ *
+ * Licensed under the Apache License, Version 2.0 the "License";
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+#include "benchmark_mode.h"
+
+namespace plugins
+{
+vkb::Flag benchmark_flag = {"benchmark", vkb::Flag::Type::FlagOnly, "Toggles using benchmark mode"};
+
+BenchmarkMode::BenchmarkMode() :
+    BenchmarkModeTags("Benchmark Mode",
+                      "Log frame averages after running an app.",
+                      {vkb::Hook::OnUpdate, vkb::Hook::OnAppStart, vkb::Hook::OnAppClose},
+					  {vkb::FlagGroup(vkb::FlagGroup::Type::Individual, true, {&benchmark_flag})})
+{
+}
+
+bool BenchmarkMode::is_active(const vkb::Parser &parser)
+{
+	return parser.contains(benchmark_flag);
+}
+
+void BenchmarkMode::init(vkb::Platform &platform, const vkb::Parser &parser)
+{
+	// Benchmark is active
+}
+
+void BenchmarkMode::on_update(float delta_time)
+{
+	elapsed_time += delta_time;
+	total_frames++;
+}
+
+void BenchmarkMode::on_app_start(const std::string &app_id)
```

Comment:
![image](https://user-images.githubusercontent.com/48598630/114415026-3bcbfc80-9ba7-11eb-8387-e0511bea9d03.png)

So here we start the app and render the first frame without calling update on any plugin. After this we calculate delta time and call on_update with the correct time of the frame. then pass 60fps to the active app. This means that all graphs shown by the app will state 60fps but the plugin itself will go off the correct time value.

I believe the justification is that when running benchmark the rendered frames must be consistent between benchmarks hence the static fps. I'm a little bit shaky on the perfect solution but this may explain the discrepancies between the actual and visual reading of fps.